src/applications/auth/containers/AuthApp.jsx

```diff
@@ -137,7 +142,11 @@ export class AuthApp extends React.Component {
   validateSession = () => {
     apiRequest('/user')
       .then(this.handleAuthSuccess)
-      .catch(this.handleAuthError);
+      .catch(
```

Comment:
This looks like it's waiting to indicate the "force-needed" cases until the subsequent API request to the `v0/user` endpoint. But the pattern of using the `auth` query parameter is something that happens on the redirect from the SAML AcS endpoint  back to the frontend. I don't see how an API response is going to set a query parameter.

The AcS callback endpoint could set this and then we could check for "auth=force-needed" in the constructor (where we currently check for "auth=fail" and set state.error). 

But I also think we need to differentiate clearly between this being in response to an auto-login attempt vs. an explicit login attempt, because that will determine whether we send the user wherever they wanted to go in the first place, or display an error.

Reply:
I agree that the query param `auth=force-needed` could be handled outside of this `validateSession` check, since it's already available before that in the callback URL. Would it make sense to move the `handleAuthForceNeeded` somewhere before `validateSession`?

Reply:
@U-DON sounds good, how does this look https://github.com/department-of-veterans-affairs/vets-website/pull/12623/commits/8156dd30bd635b07c0d5919811a24e051eba9f92

Reply:
I think this works. Since we have `auth=force-needed` instead of `auth=fail`, it would short circuit the conditional and not care about `hasSession()`. I don't know enough to say what the implications of that are (`handleAuthForceNeeded` even if you have a session?), but just wanted to mention it in case it means anything to you.

Reply:
If the auth flag is absolute, then I would even say we could directly check `if (query.auth === 'force-needed')` in the `componentDidMount` lifecycle.